Assets/MixedRealityToolkit/UX/Scripts/Tooltip/ToolTipSpawner.cs

```diff
@@ -0,0 +1,263 @@
+        public RemainBehaviorEnum RemainBehavior = RemainBehaviorEnum.Timeout;
+
+        public ToolTipConnector.FollowTypeEnum FollowType = ToolTipConnector.FollowTypeEnum.AnchorOnly;
+        public ToolTipConnector.PivotModeEnum PivotMode = ToolTipConnector.PivotModeEnum.Manual;
+        public ToolTipConnector.PivotDirectionEnum PivotDirection = ToolTipConnector.PivotDirectionEnum.North;
+        public ToolTipConnector.OrientTypeEnum PivotDirectionOrient = ToolTipConnector.OrientTypeEnum.OrientToObject;
+        public Vector3 ManualPivotDirection = Vector3.up;
+        public Vector3 ManualPivotLocalPosition = Vector3.up;
+
+        public string ToolTipText = "New Tooltip";
+
+        [SerializeField]
+        private GameObject toolTipPrefab;
+
+        public Transform Anchor;
+
+        public void OnFocusEnter(PointerSpecificEventData eventData)
+        {
+            focusEnterTime = Time.unscaledTime;
+            hasFocus = true;
+            if (toolTip == null || !toolTip.gameObject.activeSelf)
+            {
+                switch (AppearBehavior)
+                {
+                    case AppearBehaviorEnum.AppearOnFocusEnter:
+                        ShowToolTip();
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+        }
+
+        public void OnFocusExit(PointerSpecificEventData eventData)
+        {
+            focusExitTime = Time.unscaledTime;
+            hasFocus = false;
+        }
+
+        public void OnInputDown(InputEventData eventData)
+        {
+            tappedTime = Time.unscaledTime;
+            if (toolTip == null || !toolTip.gameObject.activeSelf)
+            {
+                switch (AppearBehavior)
+                {
+                    case AppearBehaviorEnum.AppearOnTap:
+                        ShowToolTip();
+                        return;
+
+                    default:
+                        break;
+                }
+            }
+        }
+
+        public void OnInputUp(InputEventData eventData)
+        {
+        }
+
+        private void ShowToolTip()
+        {
+            StartCoroutine(UpdateTooltip(focusEnterTime, tappedTime));
+        }
+
+        private IEnumerator UpdateTooltip(float focusEnterTimeOnStart, float tappedTimeOnStart)
+        {
+            if (toolTip == null)
+            {
+                GameObject toolTipGo = GameObject.Instantiate(toolTipPrefab) as GameObject;
+                toolTip = toolTipGo.GetComponent<ToolTip>();
+                toolTip.gameObject.SetActive(false);
+                toolTip.ShowBackground = showBackground;
+                toolTip.ShowOutline = showOutline;
+                toolTip.ShowConnector = showConnector;
+                toolTip.transform.position = transform.position;
+                toolTip.transform.parent = transform;
+                toolTip.ContentParentTransform.localScale = new Vector3(0.182f, 0.028f, 1.0f);
```

Comment:
we  can probably cache this vector value in a serialized private field.  What's with the magic numbers anyway?  Is this something a dev might want to customize?